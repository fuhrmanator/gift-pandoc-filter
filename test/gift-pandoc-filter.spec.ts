import { promisify } from "util";
import { exec as sysExec } from "child_process"
const exec = promisify(sysExec);

async function execPandocOnFile(fileName: string, format: "markdown" | "html" | "pdf") {
  const { stdout } = await exec(
    `pandoc -s -f markdown test/${fileName} --filter dist/index.js -t ${format}`
  );
  return String(stdout);
}

describe("GIFT Pandoc Filter", function() {
  it("should generate markdown with codeblock", async () => {
    const output = await execPandocOnFile(`example-gift-codeblock.md`, "markdown");
    expect(output).toMatchSnapshot();
  });
});