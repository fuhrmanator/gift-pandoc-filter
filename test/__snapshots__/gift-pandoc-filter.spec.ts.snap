// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIFT Pandoc Filter should generate markdown with codeblock 1`] = `
"# Exam

Text
![plantuml](https://www.plantuml.com/plantuml/svg/VL9BJiGm3DrNuWuMB60MCu8RW00XXbcCA9yD6zDwsgYfCpYEe4BS1yx1nN3AWeh5I94J-JtRptPEew9ewdqHEyTR5EmXwk39hBWg8tiJw1chhf6GkAw23m8goD4WOjqh3a5R5y4EGejSlufJ9OPDO0N7yEYu3g_HY7y9QzfWyfefZ3t1ZRYOOwxlpertrr8a2DgIHF0XTD08uPYiVBY2enux4vg4h38DXtOxjeQyfzgg2fojLd1VNjwMa_9mSZoRjlHVGSGc6zwjyvexM8OaZ7udxwshQYWhSLlToLkJZKhYLDXqX2g9YbYXxUMayBHHq03YcbQXTaALki2JqAsjoTeenqLckC1ARHwcc980HbWMsOtS666_7A9-V_Qq3-y5mD5YiPXbpZZtx3XJnPT4yiI9BVSGHtYvpjZ5zrUF8YGpEF1PK2Cu6B24-VoNChv-StqKfyIr_QaV \\"plantuml\\"){width=\\"10%\\"}

## Questions

#. Question ![plantuml](https://www.plantuml.com/plantuml/png/VL9BJiGm3DrNuWuMB60MCu8RW00XXbcCA9yD6zDwsgYfCpYEe4BS1yx1nN3AWeh5I94J-JtRptPEew9ewdqHEyTR5EmXwk39hBWg8tiJw1chhf6GkAw23m8goD4WOjqh3a5R5y4EGejSlufJ9OPDO0N7yEYu3g_HY7y9QzfWyfefZ3t1ZRYOOwxlpertrr8a2DgIHF0XTD08uPYiVBY2enux4vg4h38DXtOxjeQyfzgg2fojLd1VNjwMa_9mSZoRjlHVGSGc6zwjyvexM8OaZ7udxwshQYWhSLlToLkJZKhYLDXqX2g9YbYXxUMayBHHq03YcbQXTaALki2JqAsjoTeenqLckC1ARHwcc980HbWMsOtS666_7A9-V_Qq3-y5mD5YiPXbpZZtx3XJnPT4yiI9BVSGHtYvpjZ5zrUF8YGpEF1PK2Cu6B24-VoNChv-StqKfyIr_QaV \\"plantuml\\"){width=10%}

   A.  correct answer
   A.  wrong answer

## Another set of questions

#. Question

   A.  correct answer that is really really really really really really really really long
   A.  wrong answer


#. Question

   A.  correct answer that is really really really really really really really really long
   A.  wrong answer


#. Question

   A.  correct answer that is really really really really really really really really long
   A.  wrong answer


#. Question

   A.  correct answer that is really really really really really really really really long
   A.  wrong answer


#. Question

   A.  correct answer that is really really really really really really really really long
   A.  wrong answer


#. Question

   A.  correct answer that is really really really really really really really really long
   A.  wrong answer


#. Question

   A.  correct answer that is really really really really really really really really long
   A.  wrong answer
"
`;
